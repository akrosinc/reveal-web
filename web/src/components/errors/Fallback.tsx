import React from 'react';
import { Col, Container, Row } from 'reactstrap';
import { AN_ERROR_OCCURRED, FALLBACK_CUSTOM_ERROR_MESSAGE } from '../../configs/lang';

/**
 * Fallback
 *
 * This component is meant to be displayed in case an unrecoverable error is
 * encountered, instead of the component tree that crashed.
 *
 */
export const Fallback = () => {
  return (
    <Container
      fluid={true}
      style={{
        minHeight: '100vh',
        position: 'relative',
      }}
    >
      <Row style={{ marginTop: '40px' }} className="text-center">
        <Col sm="12" md={{ size: 4, offset: 4 }}>
          <h1>{AN_ERROR_OCCURRED}</h1>
          <div>
            <p>{FALLBACK_CUSTOM_ERROR_MESSAGE}</p>
          </div>
        </Col>
      </Row>
    </Container>
  );
};
