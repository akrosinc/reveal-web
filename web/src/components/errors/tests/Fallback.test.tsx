import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { Fallback } from '../Fallback';

describe('/components/errors/Fallback', () => {
  it('renders without crasshing', () => {
    shallow(<Fallback />);
  });

  it('render correctly', () => {
    const wrapper = mount(<Fallback />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h1').text()).toEqual('An Error Ocurred');
    expect(wrapper.find('p').text()).toEqual(
      `There has been an error. Itâ€™s been reported to the site administrators via email and should be fixed shortly. Thanks for your patience.`
    );
  });
});
