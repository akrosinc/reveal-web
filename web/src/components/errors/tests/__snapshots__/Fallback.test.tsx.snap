// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/errors/Fallback render correctly 1`] = `
<Fallback>
  <Container
    fluid={true}
    style={
      Object {
        "minHeight": "100vh",
        "position": "relative",
      }
    }
    tag="div"
  >
    <div
      className="container-fluid"
      style={
        Object {
          "minHeight": "100vh",
          "position": "relative",
        }
      }
    >
      <Row
        className="text-center"
        style={
          Object {
            "marginTop": "40px",
          }
        }
        tag="div"
      >
        <div
          className="text-center row"
          style={
            Object {
              "marginTop": "40px",
            }
          }
        >
          <Col
            md={
              Object {
                "offset": 4,
                "size": 4,
              }
            }
            sm="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-sm-12 col-md-4 offset-md-4"
            >
              <h1>
                An Error Ocurred
              </h1>
              <div>
                <p>
                  There has been an error. Itâ€™s been reported to the site administrators via email and should be fixed shortly. Thanks for your patience.
                </p>
              </div>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </Container>
</Fallback>
`;
